The problem is mainly caused by the fact that we use big icons by default.

--- framework/update/inc/f_updt_standardbar5.inc.old	2006-03-03 13:22:01.000000000 +0100
+++ framework/update/inc/f_updt_standardbar5.inc	2006-06-08 17:56:32.000000000 +0200
@@ -49,6 +49,8 @@
     
     '///<h1>Update test: Configure the Standard Bar via Tools/Customize</h1>
     '///<ul>
+
+    dim iItemsInMenu as integer
     
     '///+<li>open a new document</li>
     hCreateDocument()
@@ -58,8 +60,16 @@
 
     kontext "standardbar"
     Standardbar.OpenContextmenu()
+
     sleep( 1 )
-    hMenuselectNr( 2 )
+    '///+<li>Retrieve the number of items in the menu</li>
+    iItemsInMenu = hMenuItemGetCount()
+    
+    ' in case the toolbar does not fit into the window we have more items
+    ' (those that are invisible will be listed as context menu entries),
+    ' the menuitem (Cutomize Toolbar) is placed at position nItems - 2
+    '///+<li>Click the number of items -1, (Cutomize Toolbar)</li>
+    hMenuselectNr( iItemsInMenu - 1 )
     sleep( 1 )
     
     '///+<li>Select all tabpages ( Menu, Keyboard, Toolbars, Events )</li>
@@ -88,8 +98,16 @@
     '///+<li>Open context menu on Standardbar and select the 2nd item (Customize Toolbar)</li>
     kontext "standardbar"
     Standardbar.OpenContextmenu()
+
     sleep (1)
-    hMenuselectNr (2)
+    '///+<li>Update the number of items in the menu</li>
+    iItemsInMenu = hMenuItemGetCount()
+    
+    ' in case the toolbar does not fit into the window we have more items
+    ' (those that are invisible will be listed as context menu entries),
+    ' the menuitem (Cutomize Toolbar) is placed at position nItems - 2
+    '///+<li>Click the number of items -1, (Cutomize Toolbar)</li>
+    hMenuselectNr( iItemsInMenu - 1 )
     sleep (2)
     
     '///+<li>Close the Customize-Toolbars-Dialog with OK</li>
--- framework/tools/inc/toolbar_tools.inc.old	2006-05-19 16:28:06.000000000 +0200
+++ framework/tools/inc/toolbar_tools.inc	2006-06-08 18:34:45.000000000 +0200
@@ -118,7 +118,8 @@
     '///</ol>
     '///<u>Description</u>:
     '///<ul>
-    
+
+    dim iItemsInMenu as integer
 
     '///+<li>Create a new document</li>
     hCreateDocument()
@@ -132,9 +133,15 @@
     
     '///+<li>Open the context menu on Standardbar</li>
     Standardbar.OpenContextmenu
-    
-    '///+<li>Click on 2. entry (Cutomize Toolbar)</li>
-    hMenuselectNr (2)
+
+    '///+<li>Retrieve the number of items in the menu</li>
+    iItemsInMenu = hMenuItemGetCount()
+
+    ' in case the toolbar does not fit into the window we have more items
+    ' (those that are invisible will be listed as context menu entries),
+    ' the menuitem (Cutomize Toolbar) is placed at position nItems - 2
+    '///+<li>Click the number of items -1, (Cutomize Toolbar)</li>
+    hMenuselectNr( iItemsInMenu - 1 )
     sleep (1)
 
     kontext
@@ -203,6 +210,7 @@
     '///<ul>
     
     dim iCounter as integer
+    dim iItemsInMenu as integer
 
     '///+<li>open a new writer document</li>
     gApplication = "Writer"
@@ -217,8 +225,14 @@
     Kontext "Standardbar"
     Standardbar.OpenContextmenu()
 
-    '///+<li>select the 1. entry</li>
-    hMenuselectNr( 1 )
+    '///+<li>Retrieve the number of items in the menu</li>
+    iItemsInMenu = hMenuItemGetCount()
+
+    ' in case the toolbar does not fit into the window we have more items
+    ' (those that are invisible will be listed as context menu entries),
+    ' the menuitem (Visible Buttons) is placed at position nItems - 2
+    '///+<li>Click the number of items -2, (Visible Buttons)</li>
+    hMenuselectNr( iItemsInMenu - 2 )
     sleep (1)
 
     '///+<li>count all entries ( there must be 30 entries )</li>
@@ -266,8 +280,8 @@
     ' in case the toolbar does not fit into the window we have more items
     ' (those that are invisible will be listed as context menu entries),
     ' the menuitem (invisible items) is placed at position nItems - 3
-    '///+<li>Take the number of items -3, select the entry</li>
-    hMenuselectNr( iItemsInMenu - 3 )
+    '///+<li>Take the number of items -2, select the entry</li>
+    hMenuselectNr( iItemsInMenu - 2 )
     sleep( 1 )
     
     '///+<li>Select the provided menu position</li>
